### PR TITLE
Send single email frontend

### PR DESCRIPTION
**Pull request by En-Lun**
Send single email frontend

### CODE REVIEW

File: ClientApp/src/components/shared/EmailContentBox.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    <Typography variant="h5">Innhold</Typography>
+                        <DefaultEditor value={html} onChange={onChange} />
+                    </TableCell>
+                    <TableCell></TableCell>
+                </TableRow>
+                  
+                
+                <TableRow>
+                  <TableCell >
+                  <Button  variant="contained" endIcon={<SendIcon />} onClick={() => {sendEmailPost(); setSubjectInput("");setHtml("")}}>Send Email</Button>
```
Comment (RonHaland) [inline ClientApp/src/components/shared/EmailContentBox.tsx:98]:
```suggestion
                  <Button  variant="contained" endIcon={<SendIcon />} onClick={send()}>Send Email</Button>
```
extract this into a function

File: ClientApp/src/components/shared/EmailContentBox.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                  </TableCell>
+                  <TableCell></TableCell>
+                </TableRow>
+            </TableCell>
+      
+       
+
```
Comment (RonHaland) [inline ClientApp/src/components/shared/EmailContentBox.tsx:105]:
```suggestion
```

remove empty newlines

File: ClientApp/src/components/shared/EmailContentBox.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        const accessToken = await getAccessTokenSilently();
+        return accessToken;
+      }
+  
+    useEffect(() => {
+    getUserAccessToken().then((resp: string) => {
+        setUserAccessToken(resp);
+      });
+    });
+
```
Comment (RonHaland) [inline ClientApp/src/components/shared/EmailContentBox.tsx:48]:
```suggestion
    useEffect(() => {
      getUserAccessToken().then((resp: string) => {
        setUserAccessToken(resp);
      });
    });

```
formatting

File: ClientApp/src/components/shared/EmailContentBox.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+interface ISendSingleEmail {
+
+    open: boolean;
+    handleClose: () => void;
+    giver: GiverType;
+ }
+
+
+const SendSingleEmail: React.FC<ISendSingleEmail> = (
```
Comment (RonHaland) [inline ClientApp/src/components/shared/EmailContentBox.tsx:20]: Consider name change, either on the file itself or the component. The component and file should have the same/similar name

**Review by RonHaland — CHANGES_REQUESTED** [review]

File: ClientApp/src/components/shared/EmailContentBox.tsx (hunk 1 above)
Comment (En-Lun) [reply ClientApp/src/components/shared/EmailContentBox.tsx:98]: Fixed

File: ClientApp/src/components/shared/EmailContentBox.tsx (hunk 3 above)
Comment (En-Lun) [reply ClientApp/src/components/shared/EmailContentBox.tsx:48]: Fixed

File: ClientApp/src/components/shared/EmailContentBox.tsx (hunk 4 above)
Comment (En-Lun) [reply ClientApp/src/components/shared/EmailContentBox.tsx:20]: Fixed

**Review by MagnusVG — APPROVED** [review]

**Review by RonHaland — APPROVED** [review]

**Pull request merged**